File: Quivi.Guests.Api/Dtos/OrderItem.cs
namespace Quivi.Guests.Api.Dtos
{
    public class OrderItem : BaseOrderItem
    {
        public required IEnumerable<ModifierGroup> Modifiers { get; set; }
    }

    public class ModifierGroup
    {
        public required string Id { get; set; }
        public required IEnumerable<BaseOrderItem> SelectedOptions { get; set; }
    }

    public class BaseOrderItem
    {
        public required string Id { get; set; }
        public required string Name { get; set; }
        public decimal Amount { get; set; }
        public decimal Quantity { get; set; }
    }
}

File: quivi.guests.react.app/src/hooks/api/Dtos/orders/OrderItem.ts
export interface OrderItem extends BaseOrderItem {
    readonly modifiers?: OrderItemModifierGroup[];
}

export interface OrderItemModifierGroup {
    readonly id: string;
    readonly selectedOptions: BaseOrderItem[];
}

export interface BaseOrderItem {
    readonly id: string;
    readonly name: string;
    readonly amount: number;
    readonly quantity: number;
}
